macaroonbakery/identity.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2017 Canonical Ltd.
+# Licensed under the LGPLv3, see LICENCE file for details.
+import abc
+
+
+class Identity(object):
+    ''' Holds identity information declared in a first party caveat added when
+    discharging a third party caveat.
+    '''
+    __metaclass__ = abc.ABCMeta
+
+    @abc.abstractmethod
+    def id(self):
+        ''' Returns the id of the user
+
+        May be an opaque blob with no human meaning. An id is only considered
+        to be unique with a given domain.
+        '''
+        raise NotImplementedError('users must define id to use this base'
+                                  ' class')
+
+    @abc.abstractmethod
+    def domain(self):
+        '''Return the domain of the user.
+
+        This will be empty if the user was authenticated
+        directly with the identity provider.
+        '''
+        raise NotImplementedError('users must define domain to use this base '
+                                  'class')
+
+
+class ACLIdentity(object):
+    ''' ACLIdentity may be implemented by Identity implementions
+    to report group membership information.
+    '''
+    __metaclass__ = abc.ABCMeta
+    @abc.abstractmethod
+    def allow(self, acls):
+        '''
+        :param acls array of string acl
+        '''
+        raise NotImplementedError('users must define domain to use this base '
+                                  'class')
+
+
+class SimpleIdentity(Identity, ACLIdentity):
+    ''' A simple form of identity where the user is represented by a string.
+    '''
+    def __init__(self, user):
+        self._identity = user
+
+    def domain(self):
+        ''' Returning the empty domain.
+        '''
+        return ""
+
+    def id(self):
+        '''Return the simple identity as user
+        '''
+        return self._identity
+
+    def allow(self, acls):
+        '''Allowing the identity access to ACL members that are equal to id.
```

Comment:
Allow access to any ACL members that was equal to the user name.
?

Reply:
done.